src/ucp/wireup/wireup.c

```diff
@@ -1803,5 +1803,44 @@ unsigned ucp_ep_init_flags(const ucp_worker_h worker,
     return flags;
 }
 
+double ucp_wireup_iface_lat_distance_v1(ucp_context_h context,
+                                        const ucs_linear_func_t *latency,
+                                        const ucs_sys_dev_distance_t *distance)
+{
+    return (context->config.ext.proto_enable) ?
+        (latency->c + distance->latency) : latency->c;
+}
+
+double ucp_wireup_iface_lat_distance_v2(ucp_context_h context,
+                                        const ucs_linear_func_t *latency,
+                                        const ucs_sys_dev_distance_t *distance)
+{
+    ucs_linear_func_t lat;
+
+    if (context->config.ext.proto_enable)
+    {
+        lat    = *latency;
+        lat.c += distance->latency;
+        return ucp_tl_iface_latency(context, &lat);
+    } else {
+        return ucp_tl_iface_latency(context, latency);
+    }
+}
+
+double ucp_wireup_iface_bw_distance(ucp_context_h context,
+                                    const uct_ppn_bandwidth_t *bandwidth,
+                                    const ucs_sys_dev_distance_t *distance)
+{
+    uct_ppn_bandwidth_t bw;
+
+    if (context->config.ext.proto_enable) {
+        bw.shared    = ucs_min(bandwidth->shared, distance->bandwidth);
+        bw.dedicated = ucs_min(bandwidth->dedicated, distance->bandwidth);
+        return ucp_tl_iface_bandwidth(context, &bw);
```

Comment:
```c
double tl_bandwidth = ucp_tl_iface_bandwidth(context, bandwidth);
return context->config.ext.proto_enable ? ucs_min(tl_bandwidth, distance->bandwidth) : tl_bandwidth;
```

Reply:
Similar to my question for latency. ucp_tl_iface_bandwidth uses dedicated and shared parts. So, the above change could  result in a different value.